react/components/QuoteDetails/QuoteDetails.tsx

```diff
@@ -112,6 +117,21 @@ const QuoteDetails: FunctionComponent = () => {
   /**
    * GraphQL Queries
    */
+  const userEmail = sessionResponse?.namespaces?.profile?.email?.value
+  const { data: userData } = useQuery(CHECK_PERMISSIONS, {
+    variables: { email: userEmail },
+  })
+
+  const costCenterId: string = userData?.getUserByEmail?.[0]?.costId
+  const { data: costCenterData } = useQuery(GET_COSTCENTER_BY_ID, {
+    variables: { id: costCenterId },
+  })
+
+  const orgId: string = userData?.getUserByEmail?.[0]?.orgId
+  const { data: organizationData } = useQuery(GET_ORGANIZATION_BY_ID, {
+    variables: { id: orgId },
+  })
```

Comment:
This will make us fetch all that data every time this page is rendered, even if we end up not needing it.

It seems like we only ever need this when sending the metric, right? So it's best if we only do this data fetching at that time, within the `sendMetric` function.

In that case you won't be doing the fetching using hooks (hooks are these functions with names starting with `use`). React hooks have logic behind the scenes that cause the screen to be rerendered when the data is loading, and then again when it arrives. For your use case we should do everything behind the scenes and without causing rerenders. If these data fetches were not graphql calls you could just use `axios` for that, for example. In graphql's case you can use a simple library like [graphql-request](https://www.npmjs.com/package/graphql-request) instead.